cmd/utils/flags.go

```diff
@@ -1870,6 +1870,15 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 			chaindb := tryMakeReadOnlyDatabase(ctx, stack)
 			if rawdb.ReadCanonicalHash(chaindb, 0) != (common.Hash{}) {
 				cfg.Genesis = nil // fallback to db content
+
+				//validate genesis has PoS enabled in block 0
+				genesis, err := core.ReadGenesis(chaindb)
+				if err != nil {
+					Fatalf("Could not read genesis from database: %v", err)
+				}
+				if !genesis.Config.TerminalTotalDifficultyPassed {
```

Comment:
This check is not enough. The setting TerminalTotalDifficultyPassed configures some aspects of the client to *assume* the merge has happened, but the actual trigger for the merge is whether the total difficulty is higher than TTD. So you need to add a check here that expects `TerminalTotalDifficulty` to be set, and `Difficulty` of the genesis block to be higher than `TerminalTotalDifficulty`.

Reply:
What's interesting is that the default dev mode configuration sets the total difficulty to 0 and isTTDReached to true (so the conf is invalid, and the merge should happen on block 1).   But the simulated beacon initially sets the genesis as the finalized block and we accept it.

Reply:
This is just a weird edge-case that we don't validate in `forkchoiceUpdated` because it would never have occured on mainnet.  we just blindly accept going firmly into proof-of-stake if the beacon client indicates a finalized block.